moderntribe/tribe-common
# 116842 initial checking system

_Ref:_ [C#116841](https://central.tri.be/issues/116841)

Dependency Checker PR

Please follow the order below to review as I tried to keep the coding the same throughout.

Some notes to help explain what the changes entail:

Dev WriteUp
https://docs.google.com/document/d/1mk1rNpdRnZwAMrGQs1wY60oUhg1Zvl1WVVwWAj4Eg9U/edit#heading=h.h9mgoxmdqvd3

For ET and TEC if Dependency checker version of those plugins loads and the other plugin is before this release it will shut itself down and display a message to update the other plugin. The older plugin will fully load. This happens to prevent fatal errors as Dependency checker changes the timing of the loading and when it tries to load with an older version it crashes. 

Except for the above, ET and TEC will almost always load and only add-ons will stop loading.

Add-ons can be prevented from loading from themselves, other add-ons, and parents (et and tec)

I deprecated the min common version as checking for a minimum version of ET or TEC will do the same thing and common will load the highest version available once all are on at least dependency checker versions. Checking for a minimum ET or TEC version is a check for a minimum common version as they are always coupled together. 

The dependency admin message can display all the plugins that need to be updated, for instance if CT needs all 4 of its dependencies updated, it will be included them all in one the admin message.

The admin message also includes the minimum version a plugin needs to be updated too next to the plugin name. 

QA Grid
https://docs.google.com/spreadsheets/u/2/d/1JaplcCToJEKE5SiSOU4fcKg_4Ro1C0zviDDClUgrbSQ/edit?ouid=115512406234995825607&usp=sheets_home&ths=true

Please review in this order:

Common and TEC:
https://github.com/moderntribe/tribe-common/pull/900
https://github.com/moderntribe/the-events-calendar/pull/2353

PRO
https://github.com/moderntribe/events-pro/pull/1096

Common and ET
https://github.com/moderntribe/event-tickets/pull/1038

ET+
https://github.com/moderntribe/event-tickets-plus/pull/665

The above should all be merged before reviewing:

Filter
https://github.com/moderntribe/events-filterbar/pull/159/files

EB
https://github.com/moderntribe/events-eventbrite/pull/273

CE
https://github.com/moderntribe/events-community/pull/353

CT
https://github.com/moderntribe/events-community-tickets/pull/122
